src/samplers/gibbs.jl

```diff
@@ -71,51 +71,37 @@ function sample(model::Function, alg::Gibbs)
 
     for local_spl in spl.info[:samplers]
       if haskey(spl.info, :progress) local_spl.info[:progress] = spl.info[:progress] end
-      # dprintln(2, "Sampler stepping...")
+
       dprintln(2, "$(typeof(local_spl)) stepping...")
-      # println(varInfo)
-      if isa(local_spl.alg, Hamiltonian)
 
-        for _ = 1:local_spl.alg.n_iters
-          dprintln(2, "recording old θ...")
-          old_vi = deepcopy(varInfo)
-          is_accept, varInfo = step(model, local_spl, varInfo, i==1)
-          if ~is_accept
-            # NOTE: this might cause problem if new variables is added to VarInfo,
-            #    which will add new elements to vi.idcs etc.
-            varInfo = old_vi
-          end
-          if ~spl.alg.thin
-            samples[i_thin].value = Sample(varInfo).value
-            i_thin += 1
-          end
+      if isa(local_spl.alg, GibbsComponent)
+        if isa(local_spl.alg, PG)
+          # Clean possible wrong counter from HMC
```

Comment:
I think we can remove these code, because `fork2(vi::VarInfo)` will perform these operations anyway.